encoding should be done outside of pipeline block

beedub opened the pull request:
(reopen of https://github.com/resque/resque/pull/1243)

Redis 3.0 makes a change to return a future for some commands run inside pipeline or multi. This change allows serialization (like to_json) to read from redis and get a value, not a future.

I'm not sure how best to test this. The test I included may not even be a test at all... any ideas?

@steveklabnik

steveklabnik commented:
Hey @beedub !

First of all, apparently there's a merge conflict here, sorry about that!

Second, I agree that I'm not sure about the tests here... hm. In the last PR, it was still failing before and passing after, is this still true? it might be Good Enough for now.

beedub commented:
Fixed the merge conflict

My memory is a bit faded given how long ago it was, but from what I said in the previous PR, the test I added failed before I added my logic changes, and then passed after my logic changes (ie. encoding outside of pipeline)

steveklabnik commented on test/resque_test.rb:
```diff
@@ -9,6 +9,14 @@
     Resque.redis = @original_redis
   end
 
+  test "can push an item that depends on redis for encoding" do
```
This is causing the travis failure. It should be `it`, not `test`, like the test right below it

beedub commented:
updated the test and build now passes

steveklabnik commented:
Great! Thanks so much.
